arroyo-sql/src/avro.rs

```diff
@@ -111,3 +112,135 @@ fn to_typedef(source_name: &str, schema: &Schema) -> (TypeDef, Option<String>) {
         ),
     }
 }
+
+/// Generates code that serializes an arroyo data struct into avro
+///
+/// Note that this must align with the schemas constructed in
+/// `arroyo-formats::avro::arrow_to_avro_schema`!
+pub fn generate_serializer_item(
+    record: &Ident,
+    field: Option<&Ident>,
+    name: Option<String>,
+    td: &TypeDef,
+) -> TokenStream {
+    use DataType::*;
+    let value = match field {
+        Some(ident) => quote!(#record.#ident),
+        None => quote!(#record),
+    };
+
+    let (inner, nullable) = match td {
+        TypeDef::StructDef(def, nullable) => {
+            let fields: Vec<_> = def
+                .fields
+                .iter()
+                .map(|f| {
+                    let name = AvroFormat::sanitize_field(&match f.alias.as_ref() {
```

Comment:
What are the constraints on names here? the raw `f.name` can have math characters, e.g. `*`. Should we use the field_name which already does _ replacement on non-alpha numeric characters?